File: lab2/task_6/dynamic/JewelryShop.cpp
#include "Jewelry.cpp"
#include <iostream>

using namespace std;

class JewelryShop {
  private:
    Jewelry* jewelry = new Jewelry[3];

    int count1;
    int count2;
    int count3;

    double extraPrice;
    char name[100];

  public:
    void init(char name[], Jewelry jewelries[], int c1, int c2, int c3, double arg_extraPrice);

    void read();

    void display();

    double getFullPrice();

    Jewelry mostExpensiveJewelry();
};

void JewelryShop::init(char arg_name[], Jewelry jewelries[], int c1, int c2, int c3, double arg_extraPrice) {
      for (int i = 0; i < 3; i++) {
        jewelry[i] = jewelries[i];
      }
      
      count1 = c1;
      count2 = c2;
      count3 = c3;

      strcpy(name, arg_name);

      extraPrice = arg_extraPrice;
    };

    void JewelryShop::read() {
      cout << "Введите название магазина: ";
      cin >> name;

      cout << "Введите количество украшений 1: ";
      cin >> count1;
      jewelry[0].read();

      cout << "Введите количество украшений 2: ";
      cin >> count2;
      jewelry[1].read();

      cout << "Введите количество украшений 3: ";
      cin >> count3;
      jewelry[2].read();

      cout << "Введите стоимость доп. аксессуаров: ";
      cin >> extraPrice;
    };

void JewelryShop::display() {
  cout << "Название магазина: " << name << endl;

  cout << "Украшение 1: " << endl;
  jewelry[0].display();
  cout << "Количество: " << count1 << endl;

  cout << "Украшение 2: " << endl;
  jewelry[1].display();
  cout << "Количество: " << count2 << endl;

  cout << "Украшение 3: " << endl;
  jewelry[2].display();
  cout << "Количество: " << count3 << endl;

  cout << "Стоимость доп. аксессуаров: " << extraPrice << endl;
};

double JewelryShop::getFullPrice() {
  double fullPrice = 0;
  
  for (int i = 0; i < 3; i++) {
    fullPrice += jewelry[i].getFullPricePerGramm();
  }

  return fullPrice;
};

Jewelry JewelryShop::mostExpensiveJewelry() {
  Jewelry mostExpensive = jewelry[0];

  for (int i = 1; i < 3; i++) {
    if (jewelry[i].getFullPricePerGramm() > mostExpensive.getFullPricePerGramm()) {
      mostExpensive = jewelry[i];
    }
  }

  return mostExpensive;
};

File: lab2/task_7/cs/JewelryShop.cs
using System;

namespace Lab2
{
    public class JewelryShop
    {
        private Jewelry[] jewelry = new Jewelry[3];
        private int count1;
        private int count2;
        private int count3;

        public double extraPrice;

        private string name;

        public void Init(string arg_name, Jewelry[] jewelries, int c1, int c2, int c3, double arg_extraPrice)
        {
            jewelry = jewelries;

            count1 = c1;
            count2 = c2;
            count3 = c3;

            name = arg_name;
            extraPrice = arg_extraPrice;
        }

        public void Read()
        {
            Console.Write("Введите название магазина: ");
            name = Console.ReadLine();

            Console.Write("Введите количество украшений 1: ");
            count1 = Convert.ToInt32(Console.ReadLine());
            jewelry[0] = new Jewelry();
            jewelry[0].Read();

            Console.Write("Введите количество украшений 2: ");
            count2 = Convert.ToInt32(Console.ReadLine());
            jewelry[1] = new Jewelry();
            jewelry[1].Read();

            Console.Write("Введите количество украшений 3: ");
            count3 = Convert.ToInt32(Console.ReadLine());
            jewelry[2] = new Jewelry();
            jewelry[2].Read();

            Console.Write("Введите стоимость доп. аксессуаров: ");
            extraPrice = Convert.ToDouble(Console.ReadLine());
        }

        public void Display()
        {
            Console.WriteLine($"Название магазина: {name}");

            Console.WriteLine("Украшение 1:");
            jewelry[0].Display();
            Console.WriteLine($"Количество: {count1}");

            Console.WriteLine("Украшение 2:");
            jewelry[1].Display();
            Console.WriteLine($"Количество: {count2}");

            Console.WriteLine("Украшение 3:");
            jewelry[2].Display();
            Console.WriteLine($"Количество: {count3}");

            Console.WriteLine($"Стоимость доп. аксессуаров: {extraPrice}");
        }

        public double GetFullPrice()
        {
            double fullPrice = 0;
            
            for (int i = 0; i < count1; i++)
            {
                fullPrice += jewelry[1].GetFullPricePerGramm();
            }

            return fullPrice;
        }

        public Jewelry MostExpensiveJewelry()
        {
            Jewelry mostExpensive = jewelry[0];
            
            for (int i = 1; i < 3; i++)
            {
                if (jewelry[i].GetFullPricePerGramm() > mostExpensive.GetFullPricePerGramm())
                {
                    mostExpensive = jewelry[i];
                }
            }

            return mostExpensive;
        }
    }
}
